### PR TITLE
Report constraint violations for invalid solutions

### LINKED ISSUE #8: Return constraint violations for invalid solutions
When validating an existing solution, for those that fail, it might be helpful to know which constraints were violated and for which event/slot.

This will require the constraint generator functions to produce some form of meaningful label in addition to the constraint itself. The label can then be used to report any constraints which evaluate to `False`

### PR DESCRIPTION
**Pull request by meatballs**
Fixes #8

### CODE REVIEW

**Comment by meatballs** [comment]
This also includes all the commits in #7 as its branch was created from there. Might be worth ignoring this until #7 is merged.

**Comment by meatballs** [comment]
This makes changes that will affect any outstanding work on #2

File: conference_scheduler/resources.py [hunk 1]
```diff
@@ -61,3 +61,8 @@ class Shape(NamedTuple):
     """Represents the shape of a 2 dimensional array of events and slots"""
     events: int
     slots: int
+
+
+class Constraint(NamedTuple):
+    label: str
+    expression: bool
```
Comment (drvinceknight) [inline conference_scheduler/resources.py]: Small Technicality. Could we change `expression` to be `condition`? Technically an expression would just be one side of the constraint.

Comment (meatballs) [inline conference_scheduler/resources.py]: Good point. I'll sort that...

**Comment by drvinceknight** [comment]
Merge conflict after #7 went in.

**Pull request merged**